proposals/csharp-9.0/patterns3.md

```diff
@@ -178,11 +178,12 @@ Here, the *input type* to the second pattern is narrowed by the *type narrowing*
 1. If `P` is a type pattern, the *narrowed type* is the type of the type pattern's type.
 2. If `P` is a declaration pattern, the *narrowed type* is the type of the declaration pattern's type.
 3. If `P` is a recursive pattern that gives an explicit type, the *narrowed type* is that type.
-4. If `P` is a constant pattern where the constant is not the null constant and where the expression has no *constant expression conversion* to the *input type*, the *narrowed type* is the type of the constant.
-5. If `P` is a relational pattern where the constant expression has no *constant expression conversion* to the *input type*, the *narrowed type* is the type of the constant.
-6. If `P` is an `or` pattern, the *narrowed type* is the common type of the *narrowed type* of the subpatterns if such a common type exists. For this purpose, the common type algorithm considers only identity, boxing, and implicit reference conversions, and it considers all subpatterns of a sequence of `or` patterns (ignoring parenthesized patterns).
-7. If `P` is an `and` pattern, the *narrowed type* is the *narrowed type* of the right pattern. Moreover, the *narrowed type* of the left pattern is the *input type* of the right pattern.
-8. Otherwise the *narrowed type* of `P` is `P`'s input type.
+4. If `P` is [matched via the rules for](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/patterns.md#positional-pattern) `ITuple`, the *narrowed type* is the type `System.ITuple`.
```

Comment:
Event if the input type is not ITuple itself?
```cs
interface IMyTuple : ITuple { object Foo { get; } }

obj is IMyTuple and (1, 2) and { Foo: null } // Foo is not recognized?
```

Reply:
Confirmed we indeed get an error. two first patterns should be swapped for this to work.

Reply:
Yes. https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+BLAdvBAMwgGM4ACASQFkBPAFQFcAHAGwpHIAEBGAOgBKjfNjBw+AYQD2YZtnYIAyogBu2MlD6UmbCgG9yU4ACs4JGOQBiUqeQMBzODADc5AL7uAsACgfXAEzcPADsPno+5JFBAGzcACzk1AAURqbmhiYAlBFR4d5RBeQA+uQAvBnG5NhQVHQ67OQQuCjkSTxI5P6Zjc12VjacuIysrO6uAPTj/bbV5LhSFghmUva42ABecCgA/DmRbj4H3kA===